.ci/test.sh

```diff
@@ -75,6 +75,11 @@ if [[ $TASK == "if-else" ]]; then
     exit 0
 fi
 
+if [[ $TASK == "r-package" ]]; then
```

Comment:
Please try it to run on Azure Pipelines and with swapped compilers.

Reply:
Can you clarify this one @StrikerRUS ? I just added `r-package` task to `.vsts-ci.yml` in both Linux and macOS stages. Is that sufficient for this PR? Or would you like me to include setting up a windows environment in `test_windows.ps1`?

What do you mean by "with swapper compilers"? We use `gcc` for linux and `clang` for Mac already (https://github.com/microsoft/LightGBM/blob/master/.travis.yml#L44-L50).

Reply:
> Is that sufficient for this PR? Or would you like me to include setting up a windows environment in test_windows.ps1?

Sure! Windows part can be done in a separate PR, I guess.

> What do you mean by "with swapped compilers"?

Sorry for being not clear enough! I meant replace `gcc` with `clang` and vice versa everywhere and see what happens to make sure that R-package works with both compilers.

Reply:
I think we can pre-install R inside our docker for Azure Pipelines.
https://github.com/guolinke/lightgbm-ci-docker

Reply:
oh got it, cool!

I've created https://github.com/guolinke/lightgbm-ci-docker/pull/5 and then removed the "figure out what version of Ubuntu you are on" code in this PR as of https://github.com/microsoft/LightGBM/pull/2530/commits/1633830cb754527507dddc006a55715de391a3ab

Reply:
@StrikerRUS Right now, we're getting the following runs of `test_r_package.sh`:

* Travis - linux - `clang`
* Travis - macos - `gcc`
* Azure - linux - `gcc`
* Azure - macos - `clang`

That should be sufficient for 'test with swapped compilers', right?

Reply:
@jameslamb By `test with swapped compilers` I mean change all values to opposite ones.
If it was `clang`, it should be `gcc` and vice versa. Refer to https://github.com/guolinke/lightgbm-ci-docker/pull/3, https://github.com/microsoft/LightGBM/pull/2467/commits/2757b3f8b9dcd3d901485f2f14d99e7bfbbfdb7a as examples.

Reply:
ah you just mean "temporarily change them to get a build to be sure it works". Ok I can do that